### ISSUE TITLE
[Form] DateTimeType: view_timezone and model_timezone not working with single_text widget 

### ISSUE THREAD

**Issue by alegout** [symfony/symfony#28134]
**Symfony version(s) affected**: 4.1.2

**Description**  
In the form component, if you add a DateTimeType form element. view_timezone and model_timezone options do not work if widget option === 'single_text'.

**How to reproduce**  
Not working:
```
->add('beginDate', DateTimeType::class, [
    'widget' => 'single_text',
    'model_timezone' => 'UTC',
    'view_timezone' => date_default_timezone_get()
])
```

Working:
```
->add('beginDate', DateTimeType::class, [
    'model_timezone' => 'UTC',
    'view_timezone' => date_default_timezone_get()
])
```

Is this an expected behavior?

Thank you

**Possible Solution**  

**Additional context**

**Comment by xabbuh** [comment]
Can explain a bit more what "does not work" mean? The best would be a test case or a small example application that allows to reproduce the issue.

**Comment by alegout** [comment]
Hello @xabbuh,

Thank you for your reply. 
In my memories (not working on this part now), date is not transformed to view_timezone.
**Example:** 
in my database, date is UTC, i set view_timezone to France/Paris. It's not working, i remove `'widget' => 'single_text',` it's working. I'll confirm this after some tests. I do not have a working example, but the code i run is the same i put in this description.

Have a good day

**Comment by xabbuh** [comment]
@alegout Thank you for your details. I tried to reproduce, but for me it work as expected no matter what value I used for the `widget` option.

**Comment by alegout** [comment]
Hello @xabbuh,

Thank you for taking time to reproduce this issue.
I reproduce this morning the issue with:

- Symfony 4.1.4
- PostgreSQL 10.4

**Entity:**

```
/**
  * @ORM\Column(name="end_date", type="datetimetz", nullable=true)  
*/  
private $endDate;
```

**Form:**

```
->add('endDate', DateTimeType::class, [
  'label' => 'form.contents.end_date',
  'widget' => 'single_text',
  'required' => false,
  'model_timezone' => 'UTC',
  'view_timezone' => date_default_timezone_get() // Europe/Paris
])
```
In my table:

```
       Column        |            Type                              |
----------------+-----------------------------+
end_date            | timestamp(0) with time zone |
----------------+-----------------------------+
```

For entity 22:

```
end_date
------------------------
2019-01-24 09:47:00+00
(1 row)
```

I open my form, enddate is:

`2019-01-24 09:47:00`

Now i comment `  'widget' => 'single_text',`, i refresh the page, enddate is now:

`2019-01-24 10:47:00`

Is this helping you to reproduce?
I don't think it's a normal behavior?

Thank you very much for your help,

Alban

**Comment by xabbuh** [comment]
I tried to create a reproducer. Can you take a look at https://github.com/xabbuh/issue-28134 and modify it so that it reproduces the issue you experience?

**Comment by alegout** [comment]
Hi @xabbuh,

Thank you very much.
I installed PHP 7 on my computer to test it, it works indeed. 
I created a simple entity with a datetimetz field to try to insert a row and display it in your form but i don't have the time to finish this right now, i don't have mysql, need to install with brew and got some problems...
Will continue later. 

My date from my entity is not UTC, juste a DateTime with TimeZone '+00:00' 
My DateTime and DateTimeZone not working:

```
DateTime @1729912138 {#1358 ▼
  date: 2024-10-26 03:08:58.0 +00:00
}

DateTimeZone {#231 ▼
  timezone: +00:00
  +"timezone_type": 1
  +"timezone": "+00:00"
}
```

I'll continue in few days,

Best regards,
Alban
